### PR TITLE
Add support for /r/programming as news source

### PR DESCRIPTION
**Pull request by plorefice**
This addresses and solves #3 by introducing the `--source` option with allowed values `hn` and `reddit` (and a bit of infrastructure to easily expand to new sources).

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -38,6 +39,60 @@ type App struct {
...
+	s := geddit.NewSession(fmt.Sprintf("desktop:com.github.Bunchhieng.%s:%s", AppName, AppVersion))
+	subs, err := s.SubredditSubmissions(
+		"programming",
+		geddit.HotSubmissions,
+		geddit.ListingOptions{
+			Count: 10,
+			Limit: 10,
+		},
+	)
+	if err != nil {
```
Comment (Bunchhieng) [inline main.go]: nit: you can call `handleError(err)` here.

File: main.go [hunk 2]
```diff
@@ -38,6 +39,60 @@ type App struct {
...
+
+func (rs *RedditSource) Fetch(count int) (map[int]string, error) {
+	news := make(map[int]string)
+
+	s := geddit.NewSession(fmt.Sprintf("desktop:com.github.Bunchhieng.%s:%s", AppName, AppVersion))
+	subs, err := s.SubredditSubmissions(
+		"programming",
+		geddit.HotSubmissions,
+		geddit.ListingOptions{
+			Count: 10,
```
Comment (Bunchhieng) [inline main.go]: Should this be `Count: count`

**Review by Bunchhieng — CHANGES_REQUESTED** [review]
Nicely done! Overall looks good. Please update the count for the tab to open.

**Comment by plorefice** [comment]
@Bunchhieng fixed! :)

**Review by Bunchhieng — APPROVED** [review]

**Pull request merged**